cpp/src/strings/char_types/char_cases.h

```diff
@@ -0,0 +1,680 @@
+/*
+* Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#pragma once
+
+#include <cstdint>
+
+//
+// Each entry is the code-point's 'case opposite'
+// For example, if the code-point is an upper-case character, the entry is its lower-case counterpart.
+//
+uint16_t g_character_case_table[] = { 
```

Comment:
Hmmm, that's storing a lot of zeros... Can this be compressed somehow?

Reply:
There is an issue filed to make improvements to the case logic.